### PR TITLE
CORS-3048: Add AWS platform status to indicate user provisioned DNS

### PR DESCRIPTION
**Pull request by jhixson74**
This is a port of https://github.com/openshift/api/commit/40c7e9bd69ba859a60e7308229db8c4c8421c172 for AWS

### CODE REVIEW

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -497,6 +497,25 @@ type AWSPlatformStatus struct {
...
+	// The value is immutable after it has been set at install time.
+	// Currently, there is no way for the customer to add additional DNS entries into the cluster hosted DNS.
+	// Enabling this functionality allows the user to start their own DNS solution outside the cluster after
+	// installation is complete. The customer would be responsible for configuring this custom DNS solution,
+	// and it can be run in addition to the in-cluster DNS solution.
+	// +kubebuilder:default:="Disabled"
+	// +default="Disabled"
+	// +kubebuilder:validation:Enum="Enabled";"Disabled"
+	// +kubebuilder:validation:XValdation:rule="oldSelf == self",message="clusterHostedDNS is immutable"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="clusterHostedDNS is immutable and may only be configured during installation"
```
Comment (JoelSpeed) [inline config/v1/types_infrastructure.go:515]: These rules appear to be duplicates of one another?

Comment (jhixson74) [reply config/v1/types_infrastructure.go]: This was done as a recommendation on the GCP pr. I didn't realize I fudged it by duplication. 

Yes, ideally we would like to AWS and Azure into 4.15 if it's possible.

**Comment by sadasu** [comment]
/hold
The updates to PlatformStatus have changes to https://github.com/openshift/api/pull/1685.

**Pull request closed without merging**